pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -74,6 +81,18 @@ func NewWebhookAuthenticatorController(
 }
 
 func (c *webhookAuthenticatorController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	// TODO: remove in 4.8; this is to ensure we don't configure webhook authenticators
+	// before the oauth-apiserver revision that's capable of handling it is ready
+	// during upgrade
+	versions := c.versionGetter.GetVersions()
+	if apiserverVersion, ok := versions["oauth-apiserver"]; ok {
+		// a previous version found means this could be an upgrade, unless the version is already current
+		if expectedVersion := os.Getenv("IMAGE_OAUTH_APISERVER"); apiserverVersion != expectedVersion {
+			klog.V(4).Infof("the oauth-apiserver hasn't reported it version to be %q yet, its current version is %q", expectedVersion, apiserverVersion)
```

Comment:
create an apimachinery ratelimiter and use the `.Try` method to decide to send an event every minute we go down this path so you can tell it's happening.